mathesar/api/exceptions/database_exceptions/exceptions.py

```diff
@@ -400,12 +400,40 @@ class ExclusionViolationAPIException(MathesarAPIException):
     def __init__(
             self,
             exception,
-            message=None,
+            message="The requested update violates an exclusion constraint",
             field=None,
             details=None,
+            table=None,
             status_code=status.HTTP_500_INTERNAL_SERVER_ERROR
     ):
-        super().__init__(exception, self.error_code, message, field, details, status_code)
+        if details is None and table is not None:
+            details = {}
+            try:
+                constraint_oid = get_constraint_oid_by_name_and_table_oid(
+                    exception.orig.diag.constraint_name,
+                    table.oid,
+                    table._sa_engine
+                )
+                constraint = Constraint.objects.get(table=table, oid=constraint_oid)
+                details = {
+                    "constraint": constraint.id,
+                    "constraint_columns": [c.id for c in constraint.columns],
+                }
+            except TypeError:
+                details = {
+                    "constraint": None,
+                }
```

Comment:
When does this occur?

Reply:
I've just handled an edge case. This code will only execute in the event of a failure within the try block.

Reply:
Shouldn't that be handled using `Exception` as it a base class for all the python exceptions instead of `TypeError`?